asmcomp/amd64/emit.mlp

```diff
@@ -804,7 +804,9 @@ let emit_instr env fallthrough i =
       done;
       emit_named_text_section env.f.fun_name
   | Lentertrap ->
-      ()
+      if fp then
+        let delta = (frame_size env) - 16 (* retaddr + rbp *) in
+        I.lea (mem64 NONE delta RSP) rbp;
```

Comment:
It's hard to guess the relative precedence of `if`, `let` and semicolon, but easy to guess that of function application and subtraction.  Suggested style change:
```suggestion
      if fp then begin
        let delta = frame_size env - 16 (* retaddr + rbp *) in
        I.lea (mem64 NONE delta RSP) rbp
      end;
```